File: InterfaceAbstractConcrete_Classes/C++/Interfaz/Student.cpp
#include "Student.hpp"

Student::Student() {

}

Student::Student(int id, string name, string surname) {
    _identifier = id;
    _name = name;
    _surname = surname;
}

void Student::setIdentifier(int id) {
    _identifier = id;
}

void Student::setName(string name) {
    _name = name;
}

void Student::setSurname(string surname) {
    _surname = surname;
}

int Student::getIdentifier() {
    return _identifier;
}

string Student::getName() {
    return _name;
}

string Student::getSurname() {
    return _surname;
}

void Student::eat() {
    cout << "El estudiante come" << endl;
}

void Student::toLive() {
    cout << "El estudiante vive" << endl;
}

Student::~Student() {
    
}

File: InterfaceAbstractConcrete_Classes/Java/Interface/Student.java
package Interface;

public class Student implements Human {
    private int _identifier;
    private String _name;
    private String _surname;

    public Student(int id, String name, String surname) {
        this._identifier = id;
        this._name = name;
        this._surname = surname;
    }

    public void eat() {
        System.out.println("El estudiante come");
    }

    public void toLive() {
        System.out.println("El estudiante vive");
    }

    public void setIdentifier(int id) {
        this._identifier = id;
    }

    public void setName(String name) {
        this._name = name;
    }

    public void setSurname(String surname) {
        this._surname = surname;
    }

    public int getIdentifier() {
        return this._identifier;
    }

    public String getName() {
        return this._name;
    }

    public String getSurname() {
        return this._surname;
    }
}
